client/src/lib/browser.js

```diff
@@ -0,0 +1,27 @@
+import { useEffect, useState } from "react";
+
+export default function Browser() {
+  const [availableHeight, setAvailableHeight] = useState(calculateAvailableHeight());
+
+  useEffect(() => {
+    const handleWindowResize = () => {
+      setAvailableHeight(calculateAvailableHeight());
+    };
+
+    window.addEventListener("resize", handleWindowResize);
+
+    return () => {
+      window.removeEventListener("resize", handleWindowResize);
+    };
+  });
+
+  function calculateAvailableHeight() {
+    const appBarHeight = document.getElementById("app-bar").offsetHeight,
```

Comment:
Sometimes this function will run when the AppBar is not yet present in the dom, so `document.getElementById("app-bar").offsetHeight` would cause this line to crash, since this `document.getElementById("app-bar")` would return null. 

We can first create a variable that captures this element. If it's null, return 0. If not, then you are good to run the calculations you define below.